### PR TITLE
cppcheck5: variableScope & unreadVariable

### PR DESCRIPTION
**Pull request by 1480c1**
# Description

Fixes some variableScope and unreadVariable issues

once I get confirmation that this pr is lossless, I will squash the commits.

# Issue

# Author(s)
@1480c1

# Performance impact

- [ ] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [x] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [x] obj-2-slow
- [ ] other
- [ ] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

**Comment by EricLamVCD** [comment]
Tested obj-1 8bit and 10bit clips: 
  | Speed Deviation  | BD-Rate   Deviation | Memory   Deviation
-- | -- | -- | --
M0 - 8bit | -0.07% | 0.00% | 0.03%
M4 - 8bit | 0.41% | 0.00% | 0.01%
M8 - 8bit | 0.29% | 0.00% | 0.02%
M0 - 10bit | 0.95% | 0.00% | 0.00%
M4 - 10bit | 1.70% | 0.00% | 0.00%
M8 - 10bit | 1.87% | 0.00% | 0.00%

**Comment by 1480c1** [comment]
squashing now

File: Source/Lib/Encoder/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c [hunk 1]
```diff
@@ -5715,7 +5715,6 @@ void pme_sad_loop_kernel_avx2(uint8_t * src, // input parameter, source samples
                     ss4 = _mm256_adds_epu16(ss4, _mm256_mpsadbw_epu8(ss0, ss2, 45)); // 101 101
                     ss5 = _mm256_adds_epu16(ss5, _mm256_mpsadbw_epu8(ss1, ss2, 18)); // 010 010
                     ss6 = _mm256_adds_epu16(ss6, _mm256_mpsadbw_epu8(ss1, ss2, 63)); // 111 111
-                    p_ref += ref_stride << 2;
```
Comment (hassount) [inline Source/Lib/Encoder/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c:5718]: @ttrigui

Comment (ttrigui) [reply Source/Lib/Encoder/ASM_AVX2/EbComputeSAD_Intrinsic_AVX2.c:5718]: Valid. p_ref is being overwritten at the beginning of the loop. code being inherited from height ==8 where this was needed.

**Pull request merged**